# HG changeset patch
# User Keir Fraser <keir@xen.org>
# Date 1292415227 0
# Node ID 4e689840622fcae76b40d036ca5b3f6003f94997
# Parent  1bec63e5bcfe06af2f04d00a663df14ce53a9718
x86-64: fix restoring of hypercall arguments after trace callout

Signed-off-by: Jan Beulich <jbeulich@novell.com>
xen-unstable changeset:   22539:20c65aa19075
xen-unstable date:        Wed Dec 15 12:09:41 2010 +0000

diff -r 1bec63e5bcfe -r 4e689840622f xen/arch/x86/x86_64/entry.S
--- a/xen/arch/x86/x86_64/entry.S	Wed Dec 15 12:06:56 2010 +0000
+++ b/xen/arch/x86/x86_64/entry.S	Wed Dec 15 12:13:47 2010 +0000
@@ -171,8 +171,8 @@
         movq  UREGS_rsi+SHADOW_BYTES(%rsp),%rsi   /* Arg 2        */
         movq  UREGS_rdx+SHADOW_BYTES(%rsp),%rdx   /* Arg 3        */
         movq  UREGS_r10+SHADOW_BYTES(%rsp),%rcx   /* Arg 4        */
-        movq  UREGS_rdi+SHADOW_BYTES(%rsp),%r8    /* Arg 5        */
-        movq  UREGS_rbp+SHADOW_BYTES(%rsp),%r9    /* Arg 6        */
+        movq  UREGS_r8 +SHADOW_BYTES(%rsp),%r8    /* Arg 5        */
+        movq  UREGS_r9 +SHADOW_BYTES(%rsp),%r9    /* Arg 6        */
 #undef SHADOW_BYTES
 1:      leaq  hypercall_table(%rip),%r10
         PERFC_INCR(PERFC_hypercalls, %rax, %rbx)
